source/xatlas/xatlas.cpp

```diff
@@ -8589,7 +8589,9 @@ struct Atlas
 				Vector2 t = texcoord;
 				if (best_r) {
 					XA_DEBUG_ASSERT(options.rotateCharts);
-					swap(t.x, t.y);
+                    const float x = t.x;
+                    t.x = t.y;
+                    t.y = chartImageToPackRotated->height() - x;
```

Comment:
The center of the cell is shifted by 0.5, so we don't subtract by 1.0 to prevent violation of 
`XA_ASSERT(texcoord.x >= 0 && texcoord.y >= 0);`